lib/debugger/debugger.c

```diff
@@ -106,6 +106,10 @@ _display_source_line(LogExprNode *expr_node)
         buf[0] = 0;
       fclose(f);
     }
+  else
+    {
+      buf[0] = 0;
```

Comment:
Two lines below we have buf[strlen(buf) - 1, which will read an invalid position (buf[-1]) in this case. Maybe we should return instead.

Reply:
you are right, although the same can happen if the last line is zero length, without an NL, so that's basically a 2nd bug.

Reply:
I've added a fix for this case as well.